Plugins/MatrixPlugin/MatrixForm.cs

```diff
@@ -0,0 +1,257 @@
+            cboAttack.Enabled = false;
+            cboAttack.PopulateWithListItems(DataSource);
+            cboAttack.SelectedIndex = 0;
+            cboAttack.Visible = true;
+            cboAttack.Enabled = objThis.CurrentPerson.CanSwapAttributes;
+            cboAttack.SelectedIndexChanged += CbAttribute_SelectedIndexChanged;
+
+            cboSleaze.SelectedIndexChanged -= CbAttribute_SelectedIndexChanged;
+            cboSleaze.Enabled = false;
+            cboSleaze.PopulateWithListItems(DataSource);
+            cboSleaze.SelectedIndex = 1;
+            cboSleaze.Visible = true;
+            cboSleaze.Enabled = objThis.CurrentPerson.CanSwapAttributes;
+            cboSleaze.SelectedIndexChanged += CbAttribute_SelectedIndexChanged;
+
+            cboDP.SelectedIndexChanged -= CbAttribute_SelectedIndexChanged;
+            cboDP.Enabled = false;
+            cboDP.PopulateWithListItems(DataSource);
+            cboDP.SelectedIndex = 2;
+            cboDP.Visible = true;
+            cboDP.Enabled = objThis.CurrentPerson.CanSwapAttributes;
+            cboDP.SelectedIndexChanged += CbAttribute_SelectedIndexChanged;
+
+            cboFirewall.SelectedIndexChanged -= CbAttribute_SelectedIndexChanged;
+            cboFirewall.Enabled = false;
+            cboFirewall.PopulateWithListItems(DataSource);
+            cboFirewall.SelectedIndex = 3;
+            cboFirewall.Visible = true;
+            cboFirewall.Enabled = objThis.CurrentPerson.CanSwapAttributes;
+            cboFirewall.SelectedIndexChanged += CbAttribute_SelectedIndexChanged;
+
+            cboAttack.EndUpdate();
+            cboSleaze.EndUpdate();
+            cboDP.EndUpdate();
+            cboFirewall.EndUpdate();
+            cboAttack.ResumeLayout();
+            cboSleaze.ResumeLayout();
+            cboDP.ResumeLayout();
+            cboFirewall.ResumeLayout();
+        }
+
+        private void ListSoftware_ItemCheck(object sender, ItemCheckEventArgs e)
+        {
+            logic.ActivateSoftware(listSoftware.Items[e.Index].ToString(), e.NewValue == CheckState.Checked);
+        }
+
+        private void CbAttribute_SelectedIndexChanged(object sender, EventArgs e)
+        {
+            ComboBox from = (ComboBox)sender;
+            Action<int> funcAttributePropertySetter;
+            int oldValue;
+            if (from == cbAttack) {
+                oldValue = logic.Attack;
+                funcAttributePropertySetter = (x => logic.Attack = x);
+            } else if (from == cbSleaze) {
+                oldValue = logic.Sleaze;
+                funcAttributePropertySetter = (x => logic.Sleaze = x);
+            } else if (from == cbDataProc) {
+                oldValue = logic.DataProcessing;
+                funcAttributePropertySetter = (x => logic.DataProcessing = x);
+            } else if (from == cbFirewall) {
+                oldValue = logic.Firewall;
+                funcAttributePropertySetter = (x => logic.Firewall = x);
+            } else return;
```

Comment:
What's with the weird curly bracket notation that's different from the notation everywhere else?

Reply:
I took it from old version of the code at IHasMatrixAttributes.cs. 
(Main change in it that in original version all ComboBoxes filled with Attribute + Modifiers, my version fill only Base Attributes)
I changed writing in https://github.com/chummer5a/chummer5a/pull/4563/commits/508ab9581bb9c9f5c75463ddf88f189694de73bd